cmd/shortener/main.go

```diff
@@ -108,37 +154,129 @@ func PostShortenRequest(fileStorage *storage.FileStorage) http.HandlerFunc {
+			shortURL := utils.ShortenURL(item.OriginalURL)
+			UUID := uuid.NewString()
+
+			records = append(records, models.ShortLink{
+				UUID:        UUID,
+				ShortURL:    shortURL,
+				OriginalURL: item.OriginalURL,
+			})
+			responses = append(responses, models.BatchResponse{
+				CorrelationID: item.CorrelationID,
+				ShortURL:      conf.BaseURL + shortURL,
+			})
+		}
+
+		if err := storage.BatchInsert(context.Background(), records); err != nil {
+			http.Error(w, "Failed to save batch", http.StatusInternalServerError)
+			return
+		}
 
-	err := fileStorage.LoadFromFile()
-	if err != nil {
-		logger.Log.Error("Store not load")
+		respJSON, err := json.Marshal(responses)
+		if err != nil {
+			http.Error(w, "Encoding error", http.StatusInternalServerError)
+			return
+		}
+
+		w.Header().Set("Content-Type", "application/json")
+		w.WriteHeader(http.StatusCreated)
+		w.Write(respJSON)
+	}
+}
+
+func main() {
+	conf := config.Init()
+	if err := initLogger(conf.LogLevel); err != nil {
+		log.Fatal(err)
 	}
 
+	logger.Log.Info("Running server on", zap.String("Address", conf.Address))
+
+	strg := initStorage(conf)
+
 	r := chi.NewRouter()
+
 	r.Route("/", func(r chi.Router) {
-		r.Post("/", logger.RequestLogger(compress.GzipCompress(handler(fileStorage))))
-		r.Get("/{id}", logger.RequestLogger(compress.GzipCompress(handlerGet(fileStorage))))
+		r.Get("/{id}", logger.RequestLogger(compress.GzipCompress(handlerGet(strg))))
+		r.Post("/", logger.RequestLogger(compress.GzipCompress(handler(strg, conf))))
 		r.Route("/api/", func(r chi.Router) {
-			r.Post("/shorten", logger.RequestLogger(PostShortenRequest(fileStorage)))
+			r.Post("/shorten", logger.RequestLogger(compress.GzipCompress(PostShortenRequest(strg, conf))))
+			r.Route("/shorten/", func(r chi.Router) {
+				r.Post("/batch", logger.RequestLogger(compress.GzipCompress(HandlerShortenBatch(strg, conf))))
+			})
 		})
+		r.Get("/ping", logger.RequestLogger(compress.GzipCompress(HandlerPing(strg))))
+	})
+
+	chi.Walk(r, func(method, route string, h http.Handler, m ...func(http.Handler) http.Handler) error {
+		fmt.Printf("%-6s %s\n", method, route)
+		return nil
 	})
 
-	log.Fatal(http.ListenAndServe(config.Address, r))
+	log.Fatal(http.ListenAndServe(conf.Address, r))
 }
 
-func run() error {
-	if err := logger.Initialize(config.LogLevel); err != nil {
+func initStorage(conf *config.Config) storage.Storage {
+	var strg storage.Storage
+	var err error
+
+	store := storage.NewMapStorage()
+
+	if conf.DB != "" {
+		strg, err = storage.NewPostgresStorage(context.Background(), conf.DB)
+		if err != nil {
+			logger.Log.Error("Failed to initialize PostgreSQL storage: %v. Falling back to file storage", zap.Error(err))
+		}
+	}
+
+	if strg == nil && conf.FileStorage != "" {
+		strg = storage.NewFileStorage(conf.FileStorage, store)
+		err := strg.LoadFromFile()
+		if err != nil {
+			logger.Log.Error("Store not load")
+		}
+	}
+
+	if strg == nil {
+		strg = store
+	}
+
+	return strg
```

Comment:
Этот метод можно немного упростить.
```suggestion
	var strg storage.Storage
	var err error

	if conf.DB != "" {
		strg, err = storage.NewPostgresStorage(context.Background(), conf.DB)
		if err != nil {
			logger.Log.Error("Failed to initialize PostgreSQL storage: %v. Falling back to file storage", zap.Error(err))
		}
	} else if conf.FileStorage != "" {
		strg = storage.NewFileStorage(conf.FileStorage, storage.NewMapStorage())
		err = strg.LoadFromFile()
		if err != nil {
			logger.Log.Error("Store not load")
		}
	} else {
		strg = storage.NewMapStorage()
	}

	return strg
```

Reply:
При предложенном варианте мы не заинитим никакой storage, если например что-то случится при ините PostgresStorage